Editor/internal/preprocessing.md

```diff
@@ -0,0 +1,28 @@
+# Filter Preprocessing
+
+The GUI performs a range of preprocessing steps, prior to sending a show to fish. These steps are explained within this document.
+
+## Order of operations
+The GUI carries out these operations in the following order:
+
+1. Resolve virtual filters
+2. Instantiation of constants for default parameters of filters
+3. Perform filter optimizations
+4. Pack the show file for fish
+
+## Resolving virtual filters
+TODO explain how virtual filter resolution works.
+
+## Placement of constants
+The GUI iterates over all left empty inputs of filters and creates constant filters for these ports. It does so by creating one constant per data type and value and reuses them for each occurance.
```

Comment:
there are multiple constant filter needed for each data_type, cause the default may differ

Reply:
It has been made a bit more clear now.